api/v1/nonclusterhost_types.go

```diff
@@ -24,6 +24,9 @@ type NonClusterHostSpec struct {
 	// Location of the log ingestion point for non-cluster hosts. example: https://1.2.3.4:443
 	// +kubebuilder:validation:Pattern=`^https://.+$`
 	Endpoint string `json:"endpoint"`
+
+	// Location of the Typha endpoint for non-cluster host Felix and Typha communication. example: 5.6.7.8:5473
+	TyphaEndpoint string `json:"typhaEndpoint,omitempty"`
```

Comment:
I am open to suggestions on naming this field appropriately.

Reply:
Should it be nullable?

Reply:
I believe we can't enforce a new field to be mandatory for existing resources, so I have made it optional here. It should be a required filed moving forward though. I have improved the non-cluster host controller to validate both the `endpoint` and `typhaEndpoint` fields. If either one is in an unexpected format, the non-cluster host controller will mark its state as degraded.